exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -193,36 +186,42 @@ func validateAndSanitizeExternalLabels(externalLabels map[string]string) (map[st
 // handleScalarMetric processes data points in a single OTLP scalar metric by adding the each point as a Sample into
 // its corresponding TimeSeries in tsMap.
 // tsMap and metric cannot be nil, and metric must have a non-nil descriptor
-func (prwe *PrwExporter) handleScalarMetric(tsMap map[string]*prompb.TimeSeries, resource resourcev1.Resource, metric *otlp.Metric) error {
-	switch metric.Data.(type) {
+func (prwe *PrwExporter) handleScalarMetric(tsMap map[string]*prompb.TimeSeries, resource pdata.Resource, metric pdata.Metric) error {
+	switch metric.DataType() {
```

Comment:
Probably we can get rid of this and embed directly into the previous switch in `PushMetrics`. Let's do this in the next PR please :)

Reply:
Filed PR here: https://github.com/open-telemetry/opentelemetry-collector/pull/3261